src/app/main_modules/speech_to_text.py

```diff
@@ -44,52 +20,51 @@ def __init__(self, parent=None):
+    # Handles the event where the user closes the window with the X button.
     def closeEvent(self, event):
         event.accept()
 
 
     @pyqtSlot()
-    def ImportAudioClicked(self):
-        audioDataPath = QFileDialog.getOpenFileName(parent=self, 
-                                            caption='Open Audio Data', 
-                                            directory='./',
-                                            options=QFileDialog.DontUseNativeDialog)
-        self.audioDataPath.setText(audioDataPath[0])
-
-
-    @pyqtSlot()
-    def ImportServiceAccountClicked(self):
-        serviceAccountPath = QFileDialog.getOpenFileName(parent=self, 
-                                    caption='Open Google Service Account File', 
-                                    directory='./',
-                                    options=QFileDialog.DontUseNativeDialog)
-        self.serviceAccountPath.setText(serviceAccountPath[0])
+    def importAudioClicked(self):
+        try:
+            audioDataPath, _ = QFileDialog.getOpenFileName(parent=self, 
+                                                           caption='Import Audio Data', 
+                                                           directory='./',
+                                                           options=QFileDialog.DontUseNativeDialog)
+            if audioDataPath:
+                self.audioDataPath.setText(audioDataPath)
+        except Exception as e:
+            self.window().emitToExceptionManager(e)
 
 
     @pyqtSlot()
-    def TranscribeClicked(self):
+    def transcribeClicked(self):
         self.transcriptionResult.setText('Transcribing...')
         self.setDisabled(True)
         # The UI update is scheduled after the ending of the slot, that's why we need to force the porcessing of events.
         QApplication.processEvents()
 
-        config = {
-        'encoding' : self.encoding.currentText(),
-        'sampleRate' : self.sampleRate.value(),
-        'languageCode' : self.language.currentText(),
-        'model' : self.model.currentText(),
-        'enhanced' : self.enhanced.checkState()}
-
-        self.transcriptionResult.setText(SpeechToTextAPI.resquestTranscription(
-            self.serviceAccountPath.text(),
-            self.audioDataPath.text(),
-            config))
-
-        self.setDisabled(False)
-    
\ No newline at end of file
+        try:
+            config = {
+                'encoding' : self.encoding.currentText(),
+                'sampleRate' : self.sampleRate.value(),
+                'languageCode' : self.language.currentText(),
+                'model' : self.model.currentText(),
+                'enhanced' : self.enhanced.checkState()}
+
+            self.transcriptionResult.setText(SpeechToTextAPI.resquestTranscription(
+                self.window().settingsManager.getValue('serviceAccountPath'),
+                self.audioDataPath.text(),
+                config))
```

Comment:
The code is clearer this way:

    transcriptResponse = SpeechToTextAPI.resquestTranscription(
        self.window().settingsManager.getValue('serviceAccountPath'),
        self.audioDataPath.text(),
        config
    )
    self.transcriptionResult.setText(transcriptResponse)